docker-cpp-demos/amazonlinux-docker/Dockerfile

```diff
@@ -0,0 +1,53 @@
+# Build docker with
+# docker build -t kinesis-video-producer-sdk-cpp-amazon-linux .
+#
+FROM amazonlinux:latest
+RUN yum update -y && \
```

Comment:
I don't think `yum update -y` and `yum upgrade -y` are necessary here.

Reply:
Not sure why we need to install `vim`.

Reply:
That's absolutely fine, I love vim! I was just curious because I thought that this Dockerfile is just for running the samples. Also, I feel like editing/developing can be done outside of the container since you've mounted the host volume into the docker container. 

`yum upgrade` will do the same thing as `yum update --obsoletes`
`yum install` already ensures that yum will get the latest packages, so it'll make sure that it updates the packages first.
So, at the end, the only command that we need is only `yum install`.

Reference from `man yum`:
<img width="1366" alt="Screen Shot 2020-05-07 at 7 58 37 AM" src="https://user-images.githubusercontent.com/15654932/81291930-b8219b80-9038-11ea-9ea9-70666958447c.png">

Reply:
Removed yum update + upgrade.